test/junit/scala/math/BigDecimalTest.scala

```diff
@@ -228,4 +228,35 @@ class BigDecimalTest {
   def test_SI8970() {
     assert((0.1).## == BigDecimal(0.1).##)
   }
+
+  // Motivated by the problem of MathContext lost
+  @Test
+  def testMathContext() {
+    def testPrecision() {
+      val e = 1000
+
+      val n = BigDecimal("1.1", MC.UNLIMITED).pow(e)
+      assert(BigDecimal(1.1d, MC.UNLIMITED).pow(e) == n)
+      assert(BigDecimal.decimal(1.1d, MC.UNLIMITED).pow(e) == n)
+      assert(BigDecimal.decimal(1.1f, MC.UNLIMITED).pow(e) == n)
+      assert(BigDecimal.decimal(new BD("1.1"), MC.UNLIMITED).pow(e) == n)
+
+      val m = BigDecimal(java.lang.Double.toString(1.1f.toDouble), MC.UNLIMITED).pow(e)
+      assert(BigDecimal(1.1f, MC.UNLIMITED).pow(e) == m) // deprecated
+
+      val l = BigDecimal("11", MC.UNLIMITED).pow(e)
+      assert(BigDecimal(11, MC.UNLIMITED).pow(e) == l)
+      assert(BigDecimal.decimal(11, MC.UNLIMITED).pow(e) == l)
+    }
+
+    def testRounded() {
+      assert((BigDecimal(1.23d, new MC(3)) + BigDecimal("0.005")).rounded == BigDecimal("1.24"))
```

Comment:
This tests both the explicit number of digits (three) and the implicit rounding mode.  Can you either switch the rounding to be unambiguous (e.g. 0.007) or leave a comment about what the default rounding mode is supposed to be?